src/neptune_scale/util/logger.py

```diff
@@ -34,14 +42,37 @@ def get_logger() -> logging.Logger:
 
     ensure_style_detected()
 
-    debug = envs.get_bool(DEBUG_MODE, False)
-    logger.setLevel(logging.DEBUG if debug else logging.INFO)
+    if os.environ.get(DEBUG_MODE) is not None:
+        warnings.warn(
+            f"{DEBUG_MODE} is deprecated and will be removed in a future release. Use {LOGGER_LEVEL} instead.",
+            FutureWarning,
+        )
+
+    if logger_level := _logger_level_from_env():
+        log_format = DEBUG_FORMAT if logger_level == "DEBUG" else DEFAULT_FORMAT
 
-    stream_handler = logging.StreamHandler()
-    log_format = DEBUG_FORMAT if debug else LOG_FORMAT.format(**STYLES)
-    stream_handler.setFormatter(logging.Formatter(log_format))
-    logger.addHandler(stream_handler)
+        stream_handler = logging.StreamHandler()
+        stream_handler.setFormatter(logging.Formatter(log_format.format(**STYLES)))
+        logger.addHandler(stream_handler)
+        logger.setLevel(logger_level)
+    else:
+        logger.disabled = True
 
     logger.__neptune_scale = True  # type: ignore
 
     return logger
+
+
+_valid_levels = ("debug", "info", "warning", "error", "critical")
+
+
+def _logger_level_from_env() -> Optional[str]:
+    """Return logger level as str that can be used with logging module or None if logging is disabled"""
+    level = os.getenv(LOGGER_LEVEL, "info")
```

Comment:
Keeping `NEPTUNE_DEBUG_MODE` for compatibility doesn't make much sense if it's not actually used ;) 
I propose to read both with `NEPTUNE_LOGGER_LEVEL` taking precedence.